src/core/namespace.c

```diff
@@ -888,8 +890,13 @@ static void drop_outside_root(MountList *ml, const char *root_directory) {
 
         for (f = ml->mounts, t = ml->mounts; f < ml->mounts + ml->n_mounts; f++) {
 
-                /* ExtensionImages/Directories bases are opened in /run/systemd/unit-extensions on the host */
-                if (!IN_SET(f->mode, MOUNT_EXTENSION_IMAGE, MOUNT_EXTENSION_DIRECTORY) && !path_startswith(mount_entry_path(f), root_directory)) {
+                /* ExtensionImages/Directories bases are opened in /run/[user/xyz/]systemd/unit-extensions
+                 * on the host, and a private (invisible to the guest) tmpfs instance is mounted on
+                 * /run/[user/xyz/]systemd/unit-private-tmp as the storage backend of private /tmp and
+                 * /var/tmp. */
+                if (!IN_SET(f->mode, MOUNT_EXTENSION_IMAGE, MOUNT_EXTENSION_DIRECTORY) &&
+                    !path_startswith(mount_entry_path(f), root_directory) &&
+                    !(f->mode == MOUNT_TMPFS && endswith(mount_entry_path(f), "/systemd/unit-private-tmp"))) {
```

Comment:
Hmm, instead of doing a path prefix, please add a new MOUNT_PRIVATE_TMPFS for this and just make the decision based on this. Similar please rework rhe extension dir/image like that

Reply:
the extension ones are already set up like that, with a special enum - added MOUNT_PRIVATE_TMPFS